activesupport/lib/active_support/concern.rb

```diff
@@ -104,6 +104,13 @@ def initialize
       end
     end
 
+    class UnreachableIncludedBlock < StandardError #:nodoc:
+      def initialize
+        super "This 'included' block will never be run because the Concern was already included. " +
+          "Check your code for circular references from the Concern to the including class."
```

Comment:
Sure, I can add it. But it's becoming a very long exception message already. At what point does it become too much?

Personally, I don't see the reason why the concern name would have to be added to the message, since the whole reason we have a stack trace is to figure out where the exception comes from, and in this case it wouldn't be hard to figure out - already the second line would contain the concern name and line number. We're just duplicating info.